purchases/src/main/kotlin/com/revenuecat/purchases/virtualcurrencies/VirtualCurrencyFactory.kt

```diff
@@ -0,0 +1,39 @@
+package com.revenuecat.purchases.virtualcurrencies
+
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Builds a [VirtualCurrency] object.
+ *
+ * @throws [JSONException] If the JSON is invalid.
+ */
+internal object VirtualCurrencyFactory {
+    @Throws(JSONException::class)
+    fun buildVirtualCurrency(
+        body: JSONObject,
+    ): VirtualCurrency {
+        return body.buildVirtualCurrency()
+    }
+}
+
+internal fun JSONObject.buildVirtualCurrency(): VirtualCurrency {
+    val balance: Int = getInt(VirtualCurrencyJsonKeys.BALANCE)
+    val name: String = getString(VirtualCurrencyJsonKeys.NAME)
+    val code: String = getString(VirtualCurrencyJsonKeys.CODE)
+    val serverDescription: String? = optString(VirtualCurrencyJsonKeys.SERVER_DESCRIPTION).takeIf { it.isNotBlank() }
+
+    return VirtualCurrency(
+        balance = balance,
+        name = name,
+        code = code,
+        serverDescription = serverDescription,
+    )
+}
```

Comment:
This feels like it's too manual for a straightforward parsing of property. I assume there's some reason not to use the equivalent to iOS' `Decodable`?

Reply:
Good idea! I've streamlined the JSON encoding/decoding quite a bit by switching the coding of `VirtualCurrencies` and `VirtualCurrency` to using `@Serializable`. This both simplifies the decoding/encoding logic, but also removes the need to wrap cached virtual currencies in a `virtual_currencies` JSON object, as discussed [here](https://github.com/RevenueCat/purchases-android/pull/2466#discussion_r2185795056).

I made these changes in this commit: faf8a093325cdd92f7bde742ea3f2fd5af1d3c9a